### PR TITLE
PR-19

### PR DESCRIPTION
**Pull request by Don-Olu**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

vnet and storage lock issue

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_parameterized [hunk 1]
```diff
@@ -5,8 +5,8 @@ import groovy.json.JsonSlurper
 
 properties([
   parameters([
-    choice(name: 'ENVIRONMENT', choices:  'sbox\ndev', description: 'Environment where code should be build and deployed')
-    choice(name: 'SUBSCRIPTION', choices: 'sbox\ndev', description: 'Subscription where code should be deployed to')
+    choice(name: 'ENVIRONMENT', choices:  'sbox\ndev\ntest\nstg\nprod', description: 'Environment where code should be build and deployed')
+    choice(name: 'SUBSCRIPTION', choices: 'sbox\ndev\ntest\nstg\nprod', description: 'Subscription where code should be deployed to')
```
Comment (timja) [inline Jenkinsfile_parameterized:9]: no point doing this you can't deploy to those environments with `Jenkinsfile_parameterized` which is just for sandbox

File: prod.tfvars [hunk 2]
```diff
@@ -1,7 +1,11 @@
 sa_account_tier       = "Premium"
 sa_replication_type   = "GRS"
-vnet_address_space    = "10.50.13.128/25"
-snet01_address_prefix = "10.50.13.128/28"
-snet02_address_prefix = "10.50.13.192/26"
+vnet_address_space    = "10.101.3.0/24"
+snet01_address_prefix = "10.101.3.0/26"    #pre-video-edit-vm-prod
+snet02_address_prefix = "10.101.3.64/26"   #pre-privatelink-vm-prod
+snet01_address_prefix = "10.101.3.128/26"  #pre-bastion-prod
+snet02_address_prefix = "10.101.3.192/26"  #pre-data-gateway-prod
```
Comment (timja) [inline prod.tfvars:7]: were these allocated to you by someone?

File: prod.tfvars [hunk 3]
```diff
@@ -1,7 +1,11 @@
 sa_account_tier       = "Premium"
 sa_replication_type   = "GRS"
-vnet_address_space    = "10.50.13.128/25"
-snet01_address_prefix = "10.50.13.128/28"
-snet02_address_prefix = "10.50.13.192/26"
+vnet_address_space    = "10.101.3.0/24"
+snet01_address_prefix = "10.101.3.0/26"    #pre-video-edit-vm-prod
+snet02_address_prefix = "10.101.3.64/26"   #pre-privatelink-vm-prod
+snet01_address_prefix = "10.101.3.128/26"  #pre-bastion-prod
```
Comment (timja) [inline prod.tfvars:6]: avoid using numbers in variables just use the subnet name it'll make it easier to read then you can remove the comment

File: sa.tf [hunk 4]
```diff
@@ -67,6 +67,24 @@ module "final_storage_account" {
   common_tags = var.common_tags
 }
 
+module "streaming_storage_account" {
+  source                   = "git@github.com:hmcts/cnp-module-storage-account?ref=master"
+  env                      = var.env
+  storage_account_name     = replace("${var.product}streamingsa${var.env}", "-", "")
```
Comment (timja) [inline sa.tf:73]: avoid adding type to global resources less likely to hit length limits
```suggestion
  storage_account_name     = replace("${var.product}streaming${var.env}", "-", "")
```

File: sa.tf [hunk 5]
```diff
@@ -96,3 +124,30 @@ output "final_storage_account_primary_key" {
...
+#     private_connection_resource_id = azurerm_storage_account.storage.id
+#     is_manual_connection           = var.is_manual_connection
+#     subresource_names              = var.subResourceNames
+#   }
+
+#   private_dns_zone_group {
+#     name = lower(var.storage_account_name)
+#     private_dns_zone_ids = var.private_dns_zone_ids
+#   }
+# }
```
Comment (timja) [inline sa.tf:153]: this should be added to the module as an optional resource not added to your own code

File: vms.tf [hunk 6]
```diff
@@ -1,47 +1,84 @@
 resource "azurerm_public_ip" "pip" {
-  count               = var.num_vid_edit_vms
-  name                = "${var.product}-videditnic${count.index}pip-${var.env}"
+  # count               = var.num_vid_edit_vms
```
Comment (timja) [inline vms.tf:2]: don't leave in commented out code

File: vms.tf [hunk 7]
```diff
@@ -1,47 +1,84 @@
...
   ip_configuration {
-    name                          = "public"
-    subnet_id                     = azurerm_virtual_network.vnet.subnet.*.id[1]
+    name                          = "internal"
+    subnet_id                     = azurerm_virtual_network.vnet.subnet.*.id[0]
     private_ip_address_allocation = "Dynamic"
-    public_ip_address_id          = azurerm_public_ip.pip[count.index].id
   }
+
+  # ip_configuration {
```
Comment (timja) [inline vms.tf:26]: ?

File: vms.tf [hunk 8]
```diff
@@ -1,47 +1,84 @@
...
+#------------------------------------------------------###################
+
+resource "azurerm_bastion_host" "bastion" {
+  name                = "${var.product}-bastion-${var.env}"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+
+  ip_configuration {
+    name                          = "bastionpublic"
+    subnet_id                     = azurerm_virtual_network.vnet.subnet.*.id[2]
```
Comment (timja) [inline vms.tf:80]: don't access subnets by index, access the named one you want.
impossible to tell by looking at this code where it's going

File: vnet.tf [hunk 9]
```diff
@@ -1,18 +1,41 @@
...
+# locals {
+#   common_tags = module.ctags.common_tags
+# }
+
+# module "ctags" {
+#   source      = "git::https://github.com/hmcts/terraform-module-common-tags.git?ref=master"
+#   environment = var.env
+#   product     = var.product
+#   builtFrom   = var.builtFrom
+# }
```
Comment (timja) [inline vnet.tf:11]: not needed as you are on Jenkins

it will inject a variable `common_tags` for you to use instead

File: prod.tfvars (hunk 3 above)
Comment (Don-Olu) [reply prod.tfvars:6]: modified

File: prod.tfvars (hunk 2 above)
Comment (Don-Olu) [reply prod.tfvars:7]: Yes, @Thomas provided the /24 range

File: sa.tf (hunk 4 above)
Comment (Don-Olu) [reply sa.tf:73]: modified

File: sa.tf (hunk 5 above)
Comment (Don-Olu) [reply sa.tf:153]: To be discussed further

File: vms.tf (hunk 6 above)
Comment (Don-Olu) [reply vms.tf:2]: Modified

File: vms.tf (hunk 7 above)
Comment (Don-Olu) [reply vms.tf:26]: modified

File: vnet.tf (hunk 9 above)
Comment (Don-Olu) [reply vnet.tf:11]: modified

File: vms.tf (hunk 8 above)
Comment (Don-Olu) [reply vms.tf:80]: modified

**Review by timja — APPROVED** [review]
generally okay, better to remove commented out code though

File: sa.tf (hunk 5 above)
Comment (Don-Olu) [reply sa.tf:153]: modified

**Pull request merged**